packages/jest-phabricator-coverage/src/index.js

```diff
@@ -0,0 +1,77 @@
+/**
+ * Copyright (c) 2014, Facebook, Inc. All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ * @flow
+ */
+
+'use strict';
+
+import type {
+  AggregatedResult,
+  CoverageMap,
+} from 'types/TestResult';
+
+type PhabricatorReport = {
+  phabricatorReport: Array<Object>
+};
+
+const {formatTestResults} = require('jest-util');
+
+function summarize(coverageMap: CoverageMap, filterBy: Array<string>) {
+  let summaries = Object.create(null);
+
+  coverageMap.files().forEach(file => {
+    const covered = [];
+    const lineCoverage = coverageMap.fileCoverageFor(file).getLineCoverage();
+
+    Object.keys(lineCoverage).forEach(lineNumber => {
+      // Line numbers start at one
+      const number = (parseInt(lineNumber, 10) - 1);
+      const visited = !!lineCoverage[lineNumber];
+      covered[number] = visited ? 'C' : 'U';
+    });
+
+    for (let i = 0; i < covered.length; i++) {
+      if (!covered[i]) {
+        covered[i] = 'N';
+      }
+    }
+
+    summaries[file] = covered.join('');
+  });
+
+  if (filterBy.length) {
+    summaries = filterBy.reduce((result, file) => {
+      result[file] = summaries[file];
+      return result;
+    }, {});
```

Comment:
I wanted to gc the original huge `summaries`, that's why I used reduce here instead of creating a temporary variable.
Good point about the null prototype object, if you want to keep the reduce I might add __proto__: null, otherwise I can use forEach for readability.

I checked before the holiday and `collectCoverageFrom` doesn't seem to filter out all the `required` files, I'll give another look to how to change it so it will only loads what we need.
That's why I've add this code in the meantime, which is just to avoid passing a lot of unused data to phabricator.